### DISCUSSION TITLE
How to "pre-filter" a list on load?

### DISCUSSION THREAD

**Discussion by andrewebdev** [charmbracelet/bubbles#334] [Q&A] [answered]
The current list filtering capability is great. We do have a use case however were we have a bunch of "filter presets". Basically, these are common filters that we use on a daily basis. Rather than typing `/<filter-text>` every time, wa want the ability to quickly load the filter from a preset list. We already did the code that loads the filter text, but we have no way to actually cause the filter to update the items manually.

I noticed that there are "private" functions that will cause the list to update, but since it's private I cannot call it from our code, once we've loaded a preset.

Is there anyway for me to programatically filter the list without needing to fork it just to make a function public? Is this something that can be done via the delegate?

**Comment by taigrr** [comment]
There is a discord conversation about this here:

https://discord.com/channels/1032368467246075914/1076489986192449546

**Comment by taigrr** [comment, marked as answer]
Relevant PR #335

**Reply by andrewebdev** [reply]
Thanks! Awesome to see the quick turnaround on that.
